packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -980,11 +976,33 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     return _getTextPositionVertical(offset, verticalOffset);
   }
 
-  /// Keeping [selection]'s [TextSelection.baseOffset] fixed, move the
-  /// [TextSelection.extentOffset] down by one line.
+  // Deletes the current uncollapsed [selection].
+  void _deleteSelection(SelectionChangedCause cause) {
+    assert(_selection != null);
+    assert(_selection?.isCollapsed == false);
+
+    if (_readOnly || !_selection!.isValid || _selection!.isCollapsed) {
+      return;
+    }
+
+    final String text = textSelectionDelegate.textEditingValue.text;
+    final String textBefore = _selection!.textBefore(text);
+    final String textAfter = _selection!.textAfter(text);
+    final int cursorPosition = math.min(_selection!.start, _selection!.end);
```

Comment:
nit: IIRC it's guaranteed that `start <= end` (but not `baseOffset <= extentOffset`).